### PR TITLE
[WIP] SVG Icons

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
Bye bye font-awesome font, welcome font-awesome svgs

### Testing Instructions

screen shot 2018-06-05 at 16 40 49

### CODE REVIEW

**Comment by brianteeman** [comment]
chrome_2018-06-05_15-53-42
chrome_2018-06-05_15-55-17

**Comment by brianteeman** [comment]
https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href

**Comment by dgrammatiko** [comment]
@brianteeman what browser are you using here?

**Comment by brianteeman** [comment]
Chrome on Windows

PS are you sure this works in IE?

**Comment by dgrammatiko** [comment]
IE is very picky on svgs but there is a way to have the same icons rendered without too much hustle. But before we get to that point there are couple things that needs to be resolved first:
- Right now icons are an array in the document, I guess this needs to be either in HTMLHelper or a Library?
- Outputing the html in the template can be done either the way I'm doing it here, (unlikely) or introducing a new `<jdoc:include type="icons" />` (or something similar)
- Getting the icons from the FA repo and adding the specific IDs (plus some more inline attributes to make IE11 happy, stroke="currentColor" etc)

@mbabker @wilsonge can you point me to the right direction for the first 2 issues?

**Comment by dgrammatiko** [comment]
@brianteeman oh by the way I figured out why you're not getting the icons, your instance is in a subdirectory...

**Comment by mbabker** [comment]
@dgrammatiko life would be so much easier if we mandated Joomla can only run at root of `public_html` 🤣

**Comment by brianteeman** [comment]
Or if @dgrammatiko developed in a sub directory because he always makes the same mistake

**Comment by brianteeman** [comment]
and would it not be better to use the svg sprites instead of individual svg?

**Comment by dgrammatiko** [comment]
Sprites **packaged** by font awesome will get us back to where we were with the font. The idea here is:
- decouple from font awesome
- able to override an icon in a similar fashion to scripts and stylesheets
- Joomla is doing the packaging here (ok we're not really packaging things here but that might be a milestone 2, if we want to go this way)

**Comment by brianteeman** [comment]
Not going to ask what is wrong with being coupled to the font

Not sure if we can repackage the fonts under the licence - you would need to check that

**Comment by brianteeman** [comment]
ah - they';ve removed the restriction about bundling and packaging

**Comment by brianteeman** [comment]
still no change for me :(

**Comment by brianteeman** [comment]
Still no change :(

File: libraries/src/Document/HtmlDocument.php [hunk 1]
```diff
@@ -794,6 +798,39 @@ protected function _renderTemplate()
...
+	 * @since   4.0.0
+	 */
+	public function renderIcons()
+	{
+		$files        = [];
+
+		// Generate the file and load the stylesheet link
+		foreach ($this->icons as $key => $icon)
+		{
+			$file = HTMLHelper::image('vendor/font-awesome/' . $icon . '.svg', '', null,true, 1);
```
Comment (laoneo) [inline libraries/src/Document/HtmlDocument.php]: We made something similar in DPCalendar for J3. While doing that, an idea popped up, to first search in the template for an svg and then in vendor. Like that a template can override the core icons.

Comment (dgrammatiko) [reply libraries/src/Document/HtmlDocument.php]: exactly what is happening here as well thanks to the magic of HTMLHelper::image

File: libraries/src/Document/HtmlDocument.php [hunk 2]
```diff
@@ -794,6 +798,39 @@ protected function _renderTemplate()
...
+
+			if ($file) {
+				$matches = [];
+				preg_match('$(\/media.+)$', $file, $matches);
+				$files[] = @file_get_contents(JPATH_ROOT . $matches[1]);
```
Comment (laoneo) [inline libraries/src/Document/HtmlDocument.php]: Indeed it does. Nice one.

If you want, you can do this as an oneliner `$files[] = file_get_contents(JPATH_ROOT . substr($file, strpos($file, '/media')));`

Added the same technique now in DPCalendar.

**Comment by wilsonge** [comment]
After our 30 minute group skype discussion I'm going to close this.

**Comment by wilsonge** [comment]
Discussions outcomes:
1. Font awesome coupling in the backend
2. Inline SVGs in the frontend templates where required. Possibly coupling to font awesome on a per view basis (the backend template)

**Pull request closed without merging**